### PR TITLE
Add missing type annotations

### LINKED ISSUE #27: Missing Type Annotations
There are missing type annotations for some functions in this library.

The `typing` module does not exist on CircuitPython devices so the import needs to be wrapped in try/except to catch the error for missing import. There is an example of how that is done here:

```python
try:
    from typing import List, Tuple
except ImportError:
    pass
```

Once imported the typing annotations for the argument type(s), and return type(s) can be added to the function signature. Here is an example of a function that has had this done already:

```python
def wrap_text_to_pixels(
    string: str, max_width: int, font=None, indent0: str = "", indent1: str = ""
) -> List[str]:
```
If you are new to Git or Github we have a guide about contributing to our projects here: https://learn.adafruit.com/contribute-to-circuitpython-with-git-and-github

There is also a guide that covers our CI utilities and how to run them locally to ensure they will pass in Github Actions here: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code In particular the pages: `Sharing docs on ReadTheDocs` and `Check your code with pre-commit` contain the tools to install and commands to run locally to run the checks.

If you are attempting to resolve this issue and need help, you can post a comment on this issue and tag both @foamyguy and @kattni or reach out to us on Discord: https://adafru.it/discord in the `#circuitpython-dev` channel.

The following locations are reported by mypy to be missing type annotations:

- [ ] adafruit_trellism4.py:40
- [ ] adafruit_trellism4.py:50
- [ ] adafruit_trellism4.py:60
- [ ] adafruit_trellism4.py:70
- [ ] adafruit_trellism4.py:125
- [ ] adafruit_trellism4.py:149
- [ ] adafruit_trellism4.py:152
- [ ] adafruit_trellism4.py:231

### PR DESCRIPTION
**Pull request by tcfranks**
resolves #27 
submitted for review / comment

### CODE REVIEW

File: adafruit_trellism4.py [hunk 1]
```diff
@@ -30,14 +30,20 @@
...
+
 __version__ = "0.0.0+auto.0"
 __repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_TrellisM4.git"
 
 
 class _NeoPixelArray:
     """Creates a NeoPixel array for use in the ``TrellisM4Express`` class."""
 
-    def __init__(self, pin, *, width, height, rotation=0):
+    def __init__(self, pin: Pin, *, width: int, height: int, rotation: int = 0) -> None:
```
Comment (tekktrik) [inline adafruit_trellism4.py]: `rotation` can be `int`, but it looks like issues arise in `_calculate_pixel_offset()` if it isn't `Literal[0, 90, 180, 270]`.  Let's add `typing-extensions~=4.0` to `requirements.txt`, import `Literal` from `typing_extensions`, and then type it as such here.

File: adafruit_trellism4.py [hunk 2]
```diff
@@ -228,7 +234,7 @@ class TrellisM4Express:
             current_press = pressed
     """
 
-    def __init__(self, rotation=0):
+    def __init__(self, rotation: int = 0) -> None:
```
Comment (tekktrik) [inline adafruit_trellism4.py]: Use the same `Literal` as before here

**Review by tekktrik — CHANGES_REQUESTED** [review]

File: adafruit_trellism4.py (hunk 2 above)
Comment (tcfranks) [reply adafruit_trellism4.py]: yeah, I considered doing those like that....guess I should trust my instincts sometimes....

**Comment by tcfranks** [comment]
resubmitted with changes

**Review by tekktrik — APPROVED** [review]
Thanks!

**Pull request merged**